src/oop/special_class_method/medium/q11/MainTest.java

```diff
@@ -0,0 +1,236 @@
+        testClass.callMethod(methodName2, testInstance);
+        _assertEquals(value1 + changeValue, testClass.getFieldValue(testInstance, attributeName0),
+                "Your " + methodName2 + " method does not increase the value of " + attributeName0 + " correctly.");
+    }
+
+    @Test
+    public void changePowerIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName3, null), String.format(incorrectMethodDefinition, methodName3, className));
+        assertTrue(testClass.hasModifier(methodName3, null, "public"), String.format(incorrectModifierMessage, methodName3, className));
+        assertTrue(testClass.hasReturnType(methodName3, null, Void.TYPE), String.format(incorrectReturnType, methodName3, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctChangePowerMethod(double value1, double value2, String value3) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, double.class},
+                {value3, String.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName3, null), String.format(incorrectMethodDefinition, methodName3, className));
+        assertTrue(testClass.hasModifier(methodName3, null, "public"), String.format(incorrectModifierMessage, methodName3, className));
+        assertTrue(testClass.hasReturnType(methodName3, null, Void.TYPE), String.format(incorrectReturnType, methodName3, className));
+        testClass.callMethod(methodName3, testInstance);
+        if (value1 > changeLimit) {
+            _assertEquals(value1 - changeValue, testClass.getFieldValue(testInstance, attributeName0),
+                    "Your " + methodName3 + " method does not change the value of " + attributeName0 + " correctly.");
+        } else {
+            _assertEquals(value1 + changeValue, testClass.getFieldValue(testInstance, attributeName0),
+                    "Your " + methodName3 + " method does not change the value of " + attributeName0 + " correctly.");
+        }
+    }
+}
```

Comment:
I think it would be good to also add another test that uses the three methods together. Try to use a different input so you can test the edge cases as well.